services/mfa/mfa.go

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package mfa
+
+import (
+	b32 "encoding/base32"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/dgryski/dgoogauth"
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/mattermost/mattermost-server/services/configservice"
+	"github.com/mattermost/mattermost-server/store"
+	"github.com/mattermost/rsc/qr"
+)
+
+const (
+	MFA_SECRET_SIZE = 20
+)
+
+type Mfa struct {
+	ConfigService configservice.ConfigService
+	Store         store.Store
+}
+
+func New(configService configservice.ConfigService, store store.Store) Mfa {
+	return Mfa{configService, store}
+}
+
+func (m *Mfa) checkConfig() *model.AppError {
+	if !*m.ConfigService.Config().ServiceSettings.EnableMultifactorAuthentication {
+		return model.NewAppError("checkConfig", "mfa.license_disable.app_error", nil, "", http.StatusNotImplemented)
```

Comment:
Please could you give this a better string key and update the english i18n value.

Reply:
Sure, it comes from the original code, but of course, is a good time to change it :)

Reply:
@grundleborg Fixed, see the last commit.